### PR TITLE
add null checks when malloc() and r_anal_op_new() used

### PR DESCRIPTION
**Pull request by j123123**

### CODE REVIEW

File: libr/anal/anal.c [hunk 1]
```diff
@@ -285,7 +285,7 @@ R_API bool r_anal_set_bits(RAnal *anal, int bits) {
 
 R_API void r_anal_set_cpu(RAnal *anal, const char *cpu) {
 	free (anal->cpu);
-	anal->cpu = cpu ? strdup (cpu) : NULL;
+	anal->cpu = (cpu ? strdup (cpu) : NULL);
```
Comment (radare) [inline libr/anal/anal.c]: unnecessary parenthesis

File: libr/anal/anal.c [hunk 2]
```diff
@@ -307,8 +307,14 @@ R_API ut8 *r_anal_mask(RAnal *anal, int size, const ut8 *data, ut64 at) {
...
 	}
 
-	op = r_anal_op_new ();
-	ret = malloc (size);
+	if (!(op = r_anal_op_new ())) {
+		return NULL;
+	}
+
+	if (!(ret = malloc (size))) {
+		return NULL;
```
Comment (radare) [inline libr/anal/anal.c]: leaks op

File: libr/anal/anal.c [hunk 3]
```diff
@@ -453,7 +454,10 @@ static int nonreturn_print(void *p, const char *k, const char *v) {
 		}
 	}
 	if (!strncmp (k, "addr.", 5)) {
-		char *off = strdup (k + 5);
+		char *off;
+		if (!(off = strdup (k + 5))) {
+			return 1;
```
Comment (j123123) [inline libr/anal/anal.c]: maybe need to do something else instead ``return 1;`` in this case?

File: libr/anal/anal.c [hunk 4]
```diff
@@ -307,8 +307,15 @@ R_API ut8 *r_anal_mask(RAnal *anal, int size, const ut8 *data, ut64 at) {
...
 	}
 
-	op = r_anal_op_new ();
-	ret = malloc (size);
+	if (!(op = r_anal_op_new ())) {
+		return NULL;
+	}
+
+	if (!(ret = malloc (size))) {
+		free (op);
```
Comment (radare) [inline libr/anal/anal.c]: use r_anal_op_free

File: libr/anal/anal.c (hunk 3 above)
Comment (radare) [reply libr/anal/anal.c]: leaking op

Comment (j123123) [reply libr/anal/anal.c]: Where? I don't see op variable in this scope

**Comment by radare** [comment]
nvm

**Pull request merged**